src/base/server_type.cpp

```diff
@@ -201,6 +208,73 @@ void setup_tran_server_params_on_single_node_config ()
   prm_set_bool_value (PRM_ID_REMOTE_STORAGE, true);
 }
 
+int setup_tran_server_params_on_ha_mode ()
+{
+  assert (!HA_DISABLED ());
+
+  char *page_hosts_new_value = NULL;
+  constexpr size_t MAX_BUFSIZE = 4096;
+  int new_value_size = 0;
+
+  char ha_node_list[MAX_BUFSIZE];
+  char *str, *savep;
+
+  int port_id = prm_get_master_port_id ();
+
+  page_hosts_new_value = (char *) calloc (MAX_BUFSIZE, sizeof (char)); // free is called by sysprm_final()
+  if (page_hosts_new_value == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, MAX_BUFSIZE);
+      return ER_OUT_OF_VIRTUAL_MEMORY;
+    }
+  new_value_size = MAX_BUFSIZE;
+
+  strncpy_bufsize (ha_node_list, prm_get_string_value (PRM_ID_HA_NODE_LIST));
+
+  str = strtok_r (ha_node_list, "@", &savep); // dbname@host1:host2:...
+  str = strtok_r (NULL, " ,:", &savep);
```

Comment:
The manual says that multiple hosts are given with either `;` or `,`. Can the space also separate them?

Reply:
Oh.. you are right.
According to the manual, spaces should not be included in the delimiter. 

However, most of the legacy code that separates ha_node_list using the delimiter " :,".  (`hb_cluster_load_group_and_node_list ()`, `util_split_ha_node ()`, ...)

I will make a note of this and make sure to modify it together when the corresponding part is updated by newHA.